### ISSUE TITLE
Unstable results when running multiple times the benchmark.

### ISSUE THREAD

**Issue by H4ad** [yamiteru/isitfast#28]
Benchmark:

```js
const { suite, useTerminalCompact } = require('isitfast');

const benchmark = suite('Test', {
  gc: {
    allow: true,
  },
})

const option1 = { includePrelease: true };
const option2 = { includePrelease: true, loose: true };
const option3 = { includePrelease: true, loose: true, rtl: true };

function buildMemoKeyFromOptions(options) {
  if (options.includePrerelease === true) {
    if (options.loose === true && options.rtl === true) {
      return '1';
    }

    if (options.loose === true) {
      return '2';
    }

    if (options.rtl === true) {
      return '3';
    }

    return '4';
  } else if (options.loose === true) {
    if (options.rtl === true) {
      return '5';
    }

    return '6';
  } else if (options.rtl === true) {
    return '7';
  } else {
    return '8';
  }
}

function buildMemoKeyFromOptionsBitMasks(options) {
  return ((options.includePrerelease ? 1 << 0 : 0) | (options.loose ? 1 << 1 : 0)) + '';
}

benchmark
  .add(`Object.keys(${JSON.stringify(option1)}).join(',')`, function () {
    Object.keys(option1).join(',');
  })
  .add(`Object.keys(${JSON.stringify(option2)}).join(',')`, function () {
    Object.keys(option2).join(',');
  })
  .add(`Object.keys(${JSON.stringify(option3)}).join(',')`, function () {
    Object.keys(option3).join(',');
  });

  benchmark
  .add(`buildMemoKeyFromOptions(${JSON.stringify(option1)})`, function () {
    buildMemoKeyFromOptions(option1);
  })
  .add(`buildMemoKeyFromOptions(${JSON.stringify(option2)})`, function () {
    buildMemoKeyFromOptions(option2);
  })
  .add(`buildMemoKeyFromOptions(${JSON.stringify(option3)})`, function () {
    buildMemoKeyFromOptions(option3);
  });

  benchmark
  .add(`buildMemoKeyFromOptionsBitMasks(${JSON.stringify(option1)})`, function () {
    buildMemoKeyFromOptionsBitMasks(option1);
  })
  .add(`buildMemoKeyFromOptionsBitMasks(${JSON.stringify(option2)})`, function () {
    buildMemoKeyFromOptionsBitMasks(option2);
  })
  .add(`buildMemoKeyFromOptionsBitMasks(${JSON.stringify(option3)})`, function () {
    buildMemoKeyFromOptionsBitMasks(option3);
  });

useTerminalCompact();

benchmark.run();
```

Results:

```
Object.keys({"includePrelease":true}).join(',') 5,319,149 op/s (188 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 4,784,689 op/s (209 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 4,366,812 op/s (229 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) 17,241,379 op/s (58 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) 125,000,000 op/s (8 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) 26,315,789 op/s (38 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) 20,408,163 op/s (49 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) 16,949,153 op/s (59 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) 17,241,379 op/s (58 ns) ±1% x2,500 | 0 kB ±0% x25

h4ad:node-semver-perf-analysis/ (main) $ jsbench bench-memokey-raw-isitfast.js                                                  [20:53:40]
Object.keys({"includePrelease":true}).join(',') 5,524,862 op/s (181 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 7,633,588 op/s (131 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 3,831,418 op/s (261 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) 24,390,244 op/s (41 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) 9,090,909 op/s (110 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) 9,900,990 op/s (101 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) 90,909,091 op/s (11 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) 50,000,000 op/s (20 ns) ±1% x2,500 | 0 kB ±0% x25

h4ad:node-semver-perf-analysis/ (main) $ jsbench bench-memokey-raw-isitfast.js                                                  [20:53:43]
Object.keys({"includePrelease":true}).join(',') ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 47,619,048 op/s (21 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 7,092,199 op/s (141 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25

h4ad:node-semver-perf-analysis/ (main) $ jsbench bench-memokey-raw-isitfast.js                                                  [20:53:45]
Object.keys({"includePrelease":true}).join(',') 3,344,482 op/s (299 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 1,754,386 op/s (570 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 1,123,596 op/s (890 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) 125,000,000 op/s (8 ns) ±1% x2,500 | 0 kB ±0% x25

h4ad:node-semver-perf-analysis/ (main) $ jsbench bench-memokey-raw-isitfast.js                                                  [20:53:47]
Object.keys({"includePrelease":true}).join(',') 7,633,588 op/s (131 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 7,692,308 op/s (130 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 4,329,004 op/s (231 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) 32,258,065 op/s (31 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) 25,000,000 op/s (40 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
h4ad:node-semver-perf-analysis/ (main) $ jsbench bench-memokey-raw-isitfast.js                                                  [20:53:48]

Object.keys({"includePrelease":true}).join(',') 9,090,909 op/s (110 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true}).join(',') 8,264,463 op/s (121 ns) ±1% x2,500 | 0 kB ±0% x25
Object.keys({"includePrelease":true,"loose":true,"rtl":true}).join(',') 4,329,004 op/s (231 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true}) 33,333,333 op/s (30 ns) ±1% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptions({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
buildMemoKeyFromOptionsBitMasks({"includePrelease":true,"loose":true,"rtl":true}) ∞ op/s (0 ns) ±0% x2,500 | 0 kB ±0% x25
```

By default, I expect the results to be stable without having to tweak the default options.
The memory metric is stable but the op/s lack of stability.

**Comment by yamiteru** [comment]
After some digging around I think I've found the reason.

![image](https://user-images.githubusercontent.com/34185465/230731540-f2073cb8-f620-4296-a4a0-c80535c368bf.png)

Here you can see that I run each offset function twice and the resulting benchmark wrapper function has offset of those 2 offsets combined which is obviously wrong.

**Comment by yamiteru** [comment]
I've fixed the issue with double offset.

Although be aware that if you run too many iterations the function might become too hot and might still show infinity.

As a temporary solution I've added `options.offset.koefficient` which determines how much of the offset is actually gonna be applied. By default it's set to `1` meaning `100%` of the gathered offsets are gonna be used.

I've created an issue for this #31.
